### PR TITLE
Fix #1845: Replace Language code with icon in AudioBar [Blocked on #1872]

### LINKED ISSUE #1845: Replace Language code with icon in AudioBar
Currently Audio Bar in exploration has a language code, as shown below:
Screenshot_1600232780

We should replace the language code to [icon](https://drive.google.com/file/d/1jPRimZBxcPHeEHwlmTUcLrDcwZyVt0gl/view?usp=sharing) as shown below:
Screenshot 2020-09-16 at 10 37 42 AM

Reference: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/d74f0571-471f-4ea6-a3a5-32e135253c56/

The tests in AudioFragmentTest need to be updated (see the TODOs there).

### PR DESCRIPTION
**Pull request by MaskedCarrot**
## Explanation

Fixes #1845 
Replaced audio language text with an icon

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@MaskedCarrot there's no need to create a new PR everytime you could've tried to fix previous one and in that process you would learn something as well

**Comment by MaskedCarrot** [comment]
I will try to that from the next time :-)

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: Hi, @MaskedCarrot Thanks for the PR.
Could you confirm are the updated test cases are passing over Espresso and Robolectric?

**Comment by anandwana001** [comment]
@MaskedCarrot Congratulations on your first successful Pull Request 🎉 

I would like to add few comments which will help you in more successful and smooth contributions in Oppia-andorid. 
1. It seems like you had already completed the onboarding part, in case if you missed something you can check it out or if you find anything difficult, you can reach out to me on our gitter channel - https://github.com/oppia/oppia-android/wiki#onboarding-instructions
2. I suggest you to setup ktlint successfully at your local machine, this will defiantly help you long way - https://github.com/oppia/oppia-android/wiki#prerequisites
3. You can check this wiki page here describing about on how you can make code change properly by following good practices - https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change
4. You will be able to add Reviewers and Assignee once you can had successfully submitted 2 PR. Till then you can mention me in the PR description. 

------
1. Branch name should be something like `replace-language-code-icon-audio-bar`.
2. The PR title should follow this particular format, as per mentioned in the checklist in the PR description - `The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)`

These changes you can follow from the next PR. No need to update branch name for this PR.

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: yes the test case are passing.

File: app/src/main/res/layout/audio_fragment.xml
Comment (anandwana001) [inline app/src/main/res/layout/audio_fragment.xml]: To maintain the code format, we always use `Reformat code` option from the menu.

After reformatting this `ImageView` will be like this:

```
<ImageView
      android:id="@+id/tvAudioLanguage"
      android:layout_width="wrap_content"
      android:layout_height="wrap_content"
      android:contentDescription="@{viewModel.currentLanguageCode}"
      android:gravity="center"
      android:minWidth="48dp"
      android:minHeight="48dp"
      android:onClick="@{(v) -> audioFragment.languageSelectionClicked()}"
      android:paddingStart="8dp"
      android:paddingTop="16dp"
      android:paddingEnd="16dp"
      android:paddingBottom="16dp"
      android:src="@drawable/ic_audio_lang_24px"
      app:layout_constraintBottom_toBottomOf="parent"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintTop_toTopOf="parent" />
```

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: It seems like all of these 3 test cases are checking the same thing, is `tvAudioLanguage` visible or not. 

We can convert it to one test case mentioned below and remove the other three. 

```
@Test
  fun testAudioFragment_openFragment_showsAudioLanguageIcon() {
    launch<AudioFragmentTestActivity>(
      createHomeActivityIntent(
        PROFILE_ID_DEFAULT_AUDIO_LANGUAGE_ENGLISH
      )
    ).use {
      onView(withId(R.id.tvAudioLanguage)).check(matches(isDisplayed()))
    }
  }
```

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: These test cases are not passing for me. Could you please post the screenshot where it shows these test cases are passing for you on espresso and robolectric?

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: In case of test case not passing, please mention the below todo and annotation at the top of your test cases. 

```
  // TODO(#973): Fix AudioFragmentTest
  @Ignore
```

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (anandwana001) [inline app/src/main/res/drawable/ic_audio_lang_24px.xml]: To keep the consistency with other drawables, we can update this to:

```
<vector xmlns:android="http://schemas.android.com/apk/res/android"
  android:width="24dp"
  android:height="24dp"
  android:viewportWidth="24"
  android:viewportHeight="24">
  <path
    android:fillColor="#fff"
    android:pathData="M21.04,5.93C19.04,1.92 15.45,0.21 11.38,0.05A12.21,12.21 0,0 0,5.04 1.63,9.77 9.77,0 0,0 0.04,9.45a8.84,8.84 0,0 0,2 6.44,1.75 1.75,0 0,1 0.3,2.06c-0.47,0.89 -1,1.74 -1.56,2.59 -0.21,0.33 -0.41,0.64 -0.21,1a0.9,0.9 0,0 0,1 0.51,12.16 12.16,0 0,0 2.53,-0.7c1,-0.44 1.87,-1.09 2.81,-1.62a1.06,1.06 0,0 1,0.72 -0.09,11.52 11.52,0 0,0 9.43,-1.17A9.5,9.5 0,0 0,21.04 5.93ZM9.25,12.05a0.62,0.62 0,0 1,-0.51 -0.25,6.73 6.73,0 0,1 -0.43,-1.19c-0.11,-0.39 -0.29,-0.6 -0.74,-0.54a3.43,3.43 0,0 1,-1.06 0c-0.59,-0.1 -0.83,0.19 -1,0.7 -0.07,0.29 -0.2,0.56 -0.28,0.85s-0.29,0.56 -0.65,0.43 -0.34,-0.44 -0.23,-0.76c0.65,-1.87 1.29,-3.74 2,-5.6 0.12,-0.33 0.1,-0.83 0.59,-0.84s0.53,0.5 0.65,0.84c0.64,1.84 1.28,3.67 1.91,5.51 0.05,0.13 0.08,0.27 0.16,0.5C9.53,11.79 9.4,12.05 9.25,12.05ZM17.5,10.6a0.54,0.54 0,0 0,-0.4 0.23,7.94 7.94,0 0,1 -1.86,3.08l1.6,1c0.28,0.16 0.57,0.32 0.38,0.69s-0.51,0.33 -0.82,0.15c-0.62,-0.37 -1.23,-0.76 -1.8,-1.11l-2.31,1.17c-0.28,0.14 -0.61,0.2 -0.77,-0.18s0.1,-0.55 0.43,-0.7c0.65,-0.29 1.29,-0.63 1.92,-0.94l-1.2,-1.57c-0.2,-0.28 -0.26,-0.59 0.07,-0.78s0.54,0 0.74,0.28c0.32,0.47 0.68,0.9 1.06,1.4a6.39,6.39 0,0 0,1.66 -2.7L12.04,10.62c-0.32,0 -0.56,-0.11 -0.56,-0.48s0.24,-0.48 0.57,-0.48h2.32a3.16,3.16 0,0 1,0 -0.88c0.06,-0.21 0.27,-0.5 0.45,-0.53s0.49,0.2 0.49,0.54 0,0.53 0,0.87L17.44,9.66c0.36,0 0.77,0 0.78,0.48S17.86,10.56 17.5,10.58Z" />
  <path
    android:fillColor="#fff"
    android:pathData="M6.14,9.11L7.79,9.11L6.96,6.73C6.65,7.61 6.4,8.35 6.14,9.11Z" />
</vector>
```

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @MaskedCarrot 
Added a few changes, please let me know if you need any help in this, we can chat over gitter channel.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
PR LGTM, Thanks @MaskedCarrot 
Before the merge, could you please comment on my earlier review regarding test. 
So, I could resolve it accordingly.

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: the espresso test is failing at this line `onView(withId(R.id.tvAudioLanguage)).check(matches(isDisplayed()))`

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: We can follow the ignore annotations and todo addition, then we are good to merge this pr. Thanks.

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @MaskedCarrot

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: @anandwana001 Can you recheck this comment and if needed resolve comment.

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_audio_lang_24px.xml]: EOF is missing, add one blank line at the end.

File: app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt]: Good to go

File: app/src/main/res/layout/audio_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout/audio_fragment.xml]: This id should be updated to `audio_language_image_view`. Similarly update the test case too.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Looks good, just suggested few changes.

Thanks for the PR @MaskedCarrot

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (anandwana001) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: ah, this is something we need to track on XML linter. I will add this point to track. Thanks

**Comment by anandwana001** [comment]
@MaskedCarrot Before we review, could you please resolve the conflicts and correct the failing checks.

**Comment by MaskedCarrot** [comment]
The test in ExplorationActivityTest.kt at `testAudioWithWifi_openFractionsExploration_changeLanguage_clickNext_checkLanguageIsHinglish` is failing can you  please help @anandwana001

**Comment by anandwana001** [comment]
It's the ignore line, could you please run test on your local machine and paste the error here from the logcat?

**Comment by MaskedCarrot** [comment]
I have made the proper changes. Please review it. Sorry for taking so long to complete this.:-)

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt]: These changes are already there in the develop branch. You need to update your branch with the develop. 
https://github.com/oppia/oppia-android/blob/develop/app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt]: As we are injecting the context above we don't need to initialize it. Whenever we `@Inject` something it seems that will be provided by dagger and we don't need to initialize it.

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt]: Thanks for adding TODOs

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @MaskedCarrot Added few comments. 
Also, Why we need to update ExplorationTest here?

**Comment by anandwana001** [comment]
@MaskedCarrot  You need to update the branch with `develop` before working more on this PR as we had updated the package names.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Please update your branch with the latest `develop` branch.
Also, as per test cases in `ExplorationActivityTest`, could you please add a screenshot which shows that your test cases are passing on both espresso and robolectric?

File: app/src/main/res/layout/audio_fragment.xml
Comment (MaskedCarrot) [reply app/src/main/res/layout/audio_fragment.xml]: done

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (MaskedCarrot) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: done

File: app/src/main/res/layout/audio_fragment.xml
Comment (MaskedCarrot) [reply app/src/main/res/layout/audio_fragment.xml]: done

**Comment by MaskedCarrot** [comment]
> Please update your branch with the latest `develop` branch.

I have updated it with develop branch

> Also, as per test cases in `ExplorationActivityTest`, could you please add a screenshot which shows that your test cases are passing on both espresso and robolectric?

Screenshot from 2020-09-28 11-40-26

**Comment by anandwana001** [comment]
This PR is blocked on https://github.com/oppia/oppia-android/issues/1872
After the issue solved, the test in this PR should be updated and tested on both espresso and robolectric and good to merge. 

Thanks @MaskedCarrot 
I think by the time you can take another issue to solve or if you would like to take https://github.com/oppia/oppia-android/issues/1872 comment on the issue and let me know.

**Comment by MaskedCarrot** [comment]
the test getting ignored are the ones with reason "The ExplorationActivity takes time to finish, needs to fixed in #89." which are not related with language icon

**Comment by anandwana001** [comment]
@MaskedCarrot To make this pr full pass `testAudioFragment_openFragment_showsAudioLanguageIcon()` we need your specific test case to pass on both espresso and robolectric. 
And, as far as I know all the test cases in this file have only one issue which is regarding the data source when you run test case on robolectric, that's the reason I marked this PR blocked.

**Comment by anandwana001** [comment]
Could you please add both screenshot showing that your test cases are passing on espresso and robolectric before I review?

**Comment by MaskedCarrot** [comment]
I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this

**Comment by anandwana001** [comment]
> I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this

Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 . 

I suggest by the time feel free to pick other issues.

**Comment by MaskedCarrot** [comment]
can I take up a different issue then?

**Comment by anandwana001** [comment]
> can I take up a different issue then?

Yes you can, comment on any issue you want to work and i will assign you.

**Comment by anandwana001** [comment]
> > I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this
> 
> Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 .
> 
> I suggest by the time feel free to pick other issues.

I am able to run test cases on Robolectric as I fixed it, but didn't send the PR as it is not working on Espresso as per the #59  so @rt4914 Could you suggest what should we do in this case? 

Screenshot 2020-10-05 at 10 33 43

**Comment by rt4914** [comment]
> > > I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this
> > 
> > 
> > Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 .
> > I suggest by the time feel free to pick other issues.
> 
> I am able to run test cases on Robolectric as I fixed it, but didn't send the PR as it is not working on Espresso as per the #59 so @rt4914 Could you suggest what should we do in this case?
> 
> Screenshot 2020-10-05 at 10 33 43

@anandwana001 Does this mean that you have fixed the tests (but it is not on develop) and once you merge those tests to develop, after which this PR can be updated and will be ready to merge directly?

**Comment by anandwana001** [comment]
> > > I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this
> > 
> > 
> > Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 .
> > I suggest by the time feel free to pick other issues.
> 
> I am able to run test cases on Robolectric as I fixed it, but didn't send the PR as it is not working on Espresso as per the #59 so @rt4914 Could you suggest what should we do in this case?
> 
> Screenshot 2020-10-05 at 10 33 43



> > > > I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this
> > > 
> > > 
> > > Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 .
> > > I suggest by the time feel free to pick other issues.
> > 
> > 
> > I am able to run test cases on Robolectric as I fixed it, but didn't send the PR as it is not working on Espresso as per the #59 so @rt4914 Could you suggest what should we do in this case?
> > Screenshot 2020-10-05 at 10 33 43
> 
> @anandwana001 Does this mean that you have fixed the tests (but it is not on develop) and once you merge those tests to develop, after which this PR can be updated and will be ready to merge directly?

I can send a PR but it will not work on espresso as it is blocked on #59. I had to confirm with ben once. Though, I am trying to fix it by then.

**Comment by rt4914** [comment]
> > > > I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this
> > > 
> > > 
> > > Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 .
> > > I suggest by the time feel free to pick other issues.
> > 
> > 
> > I am able to run test cases on Robolectric as I fixed it, but didn't send the PR as it is not working on Espresso as per the #59 so @rt4914 Could you suggest what should we do in this case?
> > Screenshot 2020-10-05 at 10 33 43
> 
> > > > > I am facing a problem with that. although the test passes on GitHub. On my local machine, the test on `ExplorationActivityTest.kt ` is working fine. But when I run `AudioFragmentTest.kt` i get an error saying `Didn't find class "org.robolectric.shadows.util.DataSource` and I can't figure out a way to fix this
> > > > 
> > > > 
> > > > Yes, that is the issue I am mentioning earlier and why I want this PR to be blocked till that issue gets fixed #1872 .
> > > > I suggest by the time feel free to pick other issues.
> > > 
> > > 
> > > I am able to run test cases on Robolectric as I fixed it, but didn't send the PR as it is not working on Espresso as per the #59 so @rt4914 Could you suggest what should we do in this case?
> > > Screenshot 2020-10-05 at 10 33 43
> > 
> > 
> > @anandwana001 Does this mean that you have fixed the tests (but it is not on develop) and once you merge those tests to develop, after which this PR can be updated and will be ready to merge directly?
> 
> I can send a PR but it will not work on espresso as it is blocked on #59. I had to confirm with ben once. Though, I am trying to fix it by then.

Okay, makes sense. Can you file an issues related to fixing the test cases and it is blocked on 59 and mention it here so that we can close this PR. How does this sound?

**Comment by anandwana001** [comment]
@rt4914 

I did mention it. Let's not close the PR. We can fix it and then this PR can continue, if it took a bit longer, then we can close this PR. 

> This PR is blocked on #1872
After the issue solved, the test in this PR should be updated and tested on both espresso and robolectric and good to merge.

**Comment by rt4914** [comment]
@MaskedCarrot I am not sure what should be the next step here? You have closed this PR and re-opened so should be review it again or anything else?

**Comment by MaskedCarrot** [comment]
I accidentally closed the pr and so I reopened it. The roboelectric test needs to be fixed for `AudioFragmentTest` then this pr will be good to merge. Sorry for the inconvenience :-)

**Comment by rt4914** [comment]
> I accidentally closed the pr and so I reopened it. The roboelectric test needs to be fixed for `AudioFragmentTest` then this pr will be good to merge. Sorry for the inconvenience :-)

@MaskedCarrot

Sounds good. I have updated the title accordingly.

**Comment by anandwana001** [comment]
@MaskedCarrot I had fixed the test file, please take the latest pull and you can continue your work.

**Comment by MaskedCarrot** [comment]
@anandwana001 sorry for the delay but will it be okay if I work on this in a couple of days?

**Comment by MaskedCarrot** [comment]
@anandwana001 sorry for the delay but can I start working on this by Friday?

**Comment by anandwana001** [comment]
> @anandwana001 sorry for the delay but can I start working on this by Friday?

Sure

**Comment by anandwana001** [comment]
@MaskedCarrot any update on this?

**Comment by MaskedCarrot** [comment]
I am sorry for the delay. I tried running audioFragmentTests and its tests are failing. 
Screenshot from 2020-12-22 09-54-15

**Comment by anandwana001** [comment]
Did you update from the latest develop? @MaskedCarrot 
Some of test cases are ignored which checks pause icon, due to unavailability of fake audio library which is a separate issue.

**Comment by MaskedCarrot** [comment]
Yes my repo is up-to-date with the latest develop

**Comment by MaskedCarrot** [comment]
@anandwana001 can you please help me in fixing these test cases.

**Comment by anandwana001** [comment]
@MaskedCarrot Connecting with you on Gitter.

**Comment by MaskedCarrot** [comment]
made another pr for this issue #2308

**Pull request closed without merging**
